src/encoded/static/components/experiment.js

```diff
@@ -302,6 +302,46 @@ LibrarySubmitterComments.propTypes = {
 };
 
 
+/**
+ * Display, as one element of a "key-value" definition list, all the `construction_platform`
+ * properties in a dataset's replicates' library.
+ */
+const DatasetConstructionPlatform = ({ context }) => {
+    // Collect the library construction platforms from all the dataset replicates.
+    const constructionPlatforms = context.replicates.reduce((platforms, replicate) => {
+        // We easily have duplicates, so only add the replicate library platform if we haven't
+        // already seen it.
+        if ((replicate.library && replicate.library.construction_platform) &&
+            (platforms.findIndex(platform => platform['@id'] === replicate.library.construction_platform['@id']) === -1)) {
+            return platforms.concat(replicate.library.construction_platform);
+        }
+        return platforms;
+    }, []);
+
+    if (constructionPlatforms.length > 0) {
+        return (
+            <div data-test="constructionplatform">
+                <dt>Library construction platform</dt>
+                <dd>
+                    {constructionPlatforms.map((platform, i) =>
+                        <React.Fragment key={platform['@id']}>
+                            {i > 0 ? <span>, </span> : null}
```

Comment:
This reminds me of life under Ecmascript 5. Map creates an array. Would using .join() on the resultant array be sufficient in creating the comma separated list and clearer?

Reply:
You cannot join JSX elements. You can only join primitive elements. Other people have other ways to do this, also not involving join, and all of them very difficult to read IMO.